src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -562,18 +567,28 @@ void DiscoveryDataBase::create_participant_from_change_(
         if (ch->write_params.sample_identity().sequence_number() >
                 participant_it->second.change()->write_params.sample_identity().sequence_number())
         {
-            // Update the change related to the participant and return the old change to the pool
-            logInfo(DISCOVERY_DATABASE, "Participant updating. Marking old change to release");
-            // Update participant's change in the database, set all relevant participants ACK status to 0, and add
-            // old change to changes_to_release_.
-            update_change_and_unmatch_(ch, participant_it->second);
+            logInfo(DISCOVERY_DATABASE, "Participant already known with newer sequence number");
 
-            // If it is an update of our own server, is already in history
-            // Else, it needs to be sent in case it has unacked participants
-            if (change_guid.guidPrefix != server_guid_prefix_ &&
-                    !participant_it->second.is_acked_by_all())
+            // The change could be newer and at the same time not being an update.
+            // This happens with DATAs coming from servers, since they take their own DATAs in and out frequently,
+            // so the sequence number in `write_params` changes.
+            // To account for that, we discard the DATA if the payload is exactly the same as what wee have.
+            if (!(ch->serializedPayload == participant_it->second.change()->serializedPayload))
             {
-                add_pdp_to_send_(ch);
+                // Update the change related to the participant and return the old change to the pool
```

Comment:
This comment is not necessary, next one says the same. Also in create writer and reader